include/flatbuffers/flexbuffers.h

```diff
@@ -1629,6 +1633,231 @@ class Builder FLATBUFFERS_FINAL_CLASS {
   StringOffsetMap string_pool;
 };
 
+// Helper class to verify the integrity of a FlexBuffer
+class Verifier FLATBUFFERS_FINAL_CLASS {
+ public:
+  Verifier(const uint8_t *buf, size_t buf_len,
```

Comment:
Why not make the class just hold the `reuse_tracker` as the only dependency. Then have the `VerifyBuffer` take the buffer and length as parameters? That way a single Verifier can be used on multiple different buffers.

Reply:
We still want people to be able to verify a FlexBuffer without allocation, which means the Verifier won't own/require the vector, which in turn means a verifier is a very cheap construct (just scalars/pointers).. there's no reason to reuse them.

If anything, you could argue there should be a free-standing verification function, since all the class does is just avoid the same 3 arguments being passed to all functions recursively?

Reply:
Yeah, i was thinking a free-function would be ideal then. Makes the signatures longer but we could combine them into a small struct and just pass that around by reference.

Reply:
I'll just add a convenience free standing function.. I kind of like all the little verification helpers sitting in a class rather than passing it around by argument.